the-blue-alliance-ios/View Controllers/MyTBA/MyTBATableViewController.swift

```diff
@@ -231,105 +220,93 @@ class MyTBATableViewController<T: MyTBAEntity & MyTBAManaged, J: MyTBAModel>: TB
 
         removeNoDataView()
 
-        var request: URLSessionDataTask?
-        request = J.fetch(myTBA)() { (models, error) in
+        var reloadOperation: Operation!
+
+        var operation: MyTBAOperation!
+        operation = J.fetch(myTBA)() { [unowned self] (models, error) in
             let context = self.persistentContainer.newBackgroundContext()
             context.performChangesAndWait({
                 if let models = models as? [T.RemoteType] {
-                    let myTBAObjects = T.insert(models, in: context) as! [T]
+                    T.insert(models, in: context)
                     // Kickoff fetch for myTBA objects that don't exist
-                    for myTBAObject in myTBAObjects {
-                        let key = myTBAObject.modelKey!
+                    let operations = models.compactMap({ (myTBAObject) -> TBAKitOperation? in
+                        let key = myTBAObject.modelKey
                         switch myTBAObject.modelType {
                         case .event:
-                            self.fetchEvent(key)
+                            return self.fetchEvent(key)
                         case .team:
-                            self.fetchTeam(key)
+                            return self.fetchTeam(key)
                         case .match:
-                            self.fetchMatch(key)
+                            return self.fetchMatch(key)
                         default:
-                            break
+                            return nil
                         }
+                    })
+                    for op in operations {
+                        reloadOperation.addDependency(op)
                     }
+                    self.refreshOperationQueue.addOperations(operations, waitUntilFinished: false)
                 } else if error == nil {
                     // If we don't get any models and we don't have an error, we probably don't have any models upstream
                     context.deleteAllObjectsForEntity(entity: T.entity())
                 }
             }, saved: {
                 self.markRefreshSuccessful()
             })
-            self.removeRequest(request: request!)
         }
-        addRequest(request: request!)
+        reloadOperation = addRefreshOperations([operation]) { [weak self] in
+            DispatchQueue.main.async {
```

Comment:
This shouldn't need to be on the main thread, I think. I think it's already executed on the main thread.